app/src/main/java/org/mozilla/fenix/perf/Performance.kt

```diff
@@ -37,4 +44,22 @@ object Performance {
             activity.reportFullyDrawn()
         }
     }
+
+    fun processIntentIfPerformanceTest(intent: Intent, context: Context) {
+
+        if (intent.getBooleanExtra(HomeActivity.EXTRA_IS_PERFORMANCE_TEST, false)) {
+            val batteryStatus = context.registerReceiver(null, IntentFilter(Intent.ACTION_BATTERY_CHANGED))
+            val isPhonePlugged = batteryStatus!!.getIntExtra(BatteryManager.EXTRA_PLUGGED, -1)
+            val isAdbenabled = android.provider.Settings.Global.getInt(context.getContentResolver(),
+                                                android.provider.Settings.Global.ADB_ENABLED, 0)
+
+            if ((isPhonePlugged == BatteryManager.BATTERY_PLUGGED_USB) && (isAdbenabled == 1)) {
+                FenixOnboarding(context).finish()
+
+                for (i in 0..Settings.trackingProtectionOnboardingMaximumCount) {
```

Comment:
nit: it'd be clearer to encapsulate this in a function called something like `disableTrackingProtectionPopups` and we should add a note on why we're doing this.

It's not very self explanatory to understand what is happening and why when seeing just a for loop with an imperactive operation inside.